File: src/core/config-store.ts
import { existsSync, mkdirSync } from 'node:fs'
import { homedir } from 'node:os'
import { join } from 'node:path'

export function getContextLaneHome(): string {
  return process.env.CONTEXTLANE_HOME || join(homedir(), '.contextlane')
}

export function getRunsDir(): string {
  const dir = join(getContextLaneHome(), 'runs')
  ensureDir(dir)
  return dir
}

export function getRunDir(runId: string): string {
  const dir = join(getRunsDir(), runId)
  ensureDir(dir)
  return dir
}

export function ensureDir(dir: string): void {
  if (!existsSync(dir)) {
    mkdirSync(dir, { recursive: true })
  }
}

File: python/contextlane/core/config_store.py
import os
from pathlib import Path


def get_contextlane_home() -> str:
    return os.environ.get("CONTEXTLANE_HOME", str(Path.home() / ".contextlane"))


def get_runs_dir() -> str:
    d = Path(get_contextlane_home()) / "runs"
    d.mkdir(parents=True, exist_ok=True)
    return str(d)


def get_run_dir(run_id: str) -> str:
    d = Path(get_runs_dir()) / run_id
    d.mkdir(parents=True, exist_ok=True)
    return str(d)
